code/modules/power/geothermal/_geothermal.dm

```diff
@@ -1,63 +1,192 @@
-	layer = TURF_LAYER + 0.01
+	anchored   = TRUE
+	layer      = TURF_LAYER + 0.01
+	level      = 1 //Goes under floor/plating
+	var/particles/geyser_steam/steamfx
 
-/obj/effect/geyser/Initialize()
+/obj/effect/geyser/Initialize(ml)
 	. = ..()
 	if(prob(50))
 		var/matrix/M = matrix()
 		M.Scale(-1, 1)
 		transform = M
 	set_extension(src, /datum/extension/geothermal_vent)
+	steamfx = new //Prepare our FX
+
+/obj/effect/geyser/proc/do_spout()
+	particles = steamfx
+	particles.spawning = initial(particles.spawning)
+	spawn(6 SECONDS)
+		particles.spawning = 0
+
+/obj/effect/geyser/explosion_act(severity)
+	. = ..()
+	if(!QDELETED(src) && prob(100 - (25 * severity)))
+		physically_destroyed()
+
+/obj/effect/geyser/hide(hide)
+	var/datum/extension/geothermal_vent/E = get_extension(src, /datum/extension/geothermal_vent)
+	if(istype(E))
+		E.set_obstructed(hide)
+	. = ..()
+	update_icon()
+
+//////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////
+/obj/effect/geyser/underwater
+	desc = "A crack in the ocean floor that occasionally vents gouts of superheated water and steam."
+
+/obj/effect/geyser/underwater/Initialize(ml)
+	. = ..()
 	for(var/turf/exterior/seafloor/T in RANGE_TURFS(loc, 5))
 		var/dist = get_dist(loc, T)-1
 		if(prob(100 - (dist * 20)))
 			T = T.ChangeTurf(/turf/exterior/mud)
 		if(prob(50 - (dist * 10)))
 			new /obj/random/seaweed(T)
 
-/obj/effect/geyser/explosion_act(severity)
-	. = ..()
-	if(!QDELETED(src) && prob(100 - (25 * severity)))
-		physically_destroyed()
+/obj/effect/geyser/underwater/do_spout()
+	//Do some bubbles or something
+
+//////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////
+/obj/machinery/geothermal/buildable
+	anchored                  = FALSE
+	uncreated_component_parts = null
+	stock_part_presets        = null
 
 /obj/machinery/geothermal
-	icon = 'icons/obj/machines/power/geothermal.dmi'
-	icon_state = "geothermal-base"
-	var/tmp/neighbors = 0
-	var/tmp/current_pressure = 0
-	var/efficiency = 0.5
+	name                           = "geothermal generator"
+	icon                           = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state                     = "geothermal-base"
+	density                        = TRUE
+	anchored                       = TRUE
+	waterproof                     = TRUE
+	interact_offline               = TRUE
+	stat_immune                    = NOSCREEN | NOINPUT | NOPOWER
+	core_skill                     = SKILL_ENGINES
+	required_interaction_dexterity = DEXTERITY_SIMPLE_MACHINES
+	construct_state                = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts      = list(
+		/obj/item/stock_parts/power/terminal = 1
+	)
+	stock_part_presets             = list(
+		/decl/stock_part_preset/terminal_setup/offset_dir,
+	)
+	base_type = /obj/machinery/geothermal/buildable
```

Comment:
I think you will accomplish the same effect by setting `uncreated_component_parts = null`, adding the unbuildable terminal to `additional_spawn_components`, and getting rid of this subtype. That also has the advantage that you can use a _buildable_ terminal component, which is more player-friendly.

Reply:
I think the buildable type was kind of a leftover I forgot about. I just tweaked the base type instead.